tremor-script/src/ast/raw.rs

```diff
@@ -154,6 +158,134 @@ impl<'script> ScriptRaw<'script> {
     }
 }
 
+#[derive(Debug, PartialEq, Serialize, Clone, Copy)]
+pub enum BytesDataType {
+    SignedInteger,
+    UnsignedInteger,
+    Binary,
+}
+
+impl Default for BytesDataType {
+    fn default() -> Self {
+        BytesDataType::UnsignedInteger
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Clone, Copy)]
+pub enum Endian {
+    Little,
+    Big,
+}
+
+impl Default for Endian {
+    fn default() -> Self {
+        Endian::Big
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Clone)]
+pub struct BytesPartRaw<'script> {
+    pub start: Location,
+    pub end: Location,
+    pub data: ImutExprRaw<'script>,
+    pub data_type: IdentRaw<'script>,
+    pub bits: Option<i64>,
+}
+impl_expr!(BytesPartRaw);
+
+impl<'script> Default for BytesPartRaw<'script> {
+    fn default() -> Self {
+        BytesPartRaw {
+            start: Location::default(),
+            end: Location::default(),
+            data: ImutExprRaw::Literal(LiteralRaw::default()),
+            data_type: IdentRaw::default(),
+            bits: None,
+        }
+    }
+}
+impl<'script> Upable<'script> for BytesPartRaw<'script> {
+    type Target = BytesPart<'script>;
+    // We allow this for casting the bits
+    #[allow(clippy::cast_sign_loss)]
+    fn up<'registry>(self, helper: &mut Helper<'script, 'registry>) -> Result<Self::Target> {
+        let data_type: Vec<&str> = self.data_type.id.split('-').collect();
+        let (data_type, endianess) = match data_type.as_slice() {
+            ["binary"] => (BytesDataType::Binary, Endian::Big),
+            []
+            | [""]
+            | ["integer"]
+            | ["big"]
+            | ["unsigned"]
+            | ["unsigned", "integer"]
+            | ["big", "integer"]
```

Comment:
I would prefer to stay with `big` and `little` and the order it mirrors the Erlang binaries and it feels wrong to implement 90% and then just skip a little thing.

This way there is a refference where there is a familiar for some people instead of new for some and confusing for others.

Reply:
i would say the audience that exactly knows erlang binary syntax is rather small, but i get your point. fine by me.